packages/jsx-parser/src/index.ts

```diff
@@ -1,15 +1,12 @@
 import { JSXElement, createMemo } from "solid-js";
 
-export function createJSXParser(id: string = "solid-parser") {
+export function createJSXParser<TTokens = {}>(id: string = "solid-parser") {
   const $TOKEN = Symbol(id);
 
-  function createToken<
-    TProps extends { [key: string]: any },
-    TToken extends { [key: string]: any } & { id: string }
-  >(
+  function createToken<TProps extends { [key: string]: any }, TToken extends TTokens>(
```

Comment:
is there a way you can make `TToken` fit the union `TTokens` without `extends`?

``` ts
type GreenToken = {id: 'green'}

const {createToken, childrenTokens} = createJSXParser<GreenToken>('color-parser')

const Green = createToken<Props>(props => ({id: 'green', extra: 'key'}) 
  // no type-error, since {props, id: 'green', extra: 'key'} extends {id: 'green'}

```